src/Stratis.Bitcoin.Features.Consensus.Tests/TimeSyncBehaviorTest.cs

```diff
@@ -28,173 +28,283 @@ public class TimeSyncBehaviorTest
+        /// </summary>
+        public bool IsInbound { get; }
 
-            adjustedTime = dateTimeProvider.GetAdjustedTime();
-            normalTime = dateTimeProvider.GetUtcNow();
-            diff = adjustedTime - normalTime;
-            expectedDiffMs = allSamples.Median();
-            Assert.True(Math.Abs(diff.TotalMilliseconds) < Math.Abs(expectedDiffMs) + TimeEpsilonMs);
+        /// <summary>
+        /// Expectation of whether the sample is used or not.
+        /// </summary>
+        public bool ExpectedIsUsed { get; }
+        
+        
+        /// <summary>
+        /// >Expectation of whether the warning has been set to on.
+        /// </summary>
+        public bool ExpectedIsWarningOn { get; }
+        
+        /// <summary>
+        /// 
+        /// </summary>
+        public bool ExpectedIsSyncOff { get; }
+        
+        /// <summary>
+        /// Expectation that the time offset will be set to this value.
+        /// </summary>
+        public int ExpectedTimeOffsetMs { get; }
+        
+        /// <summary>
+        /// The actual inout time offset from the sample.
+        /// </summary>
+        public TimeSpan InputTimeOffset { get; }
+        
+        /// <summary>
+        /// The IP Address of the sample.
+        /// </summary>
+        public IPAddress PeerIpAddress { get; }
+
+        /// <summary>
+        /// Private constructor, forcing the factory methods Inbound and Outbound to be used.
+        /// </summary>
+        /// <param name="isInbound">Input isInbound flag, with false meaning outbound.</param>
+        /// <param name="expectedIsUsed">Expectation of whether the sample is used or not.</param>
+        /// <param name="expectedIsWarningOn">Expectation of whether the warning has been set to on.</param>
+        /// <param name="expectedIsSyncOff">Expectation of whether the sync has been set to off.</param>
+        /// <param name="expectedTimeOffsetMs">Expectation that the time offset will be set to this value.</param>
+        /// <param name="inputTimeOffset">The actual inout time offset from the sample.</param>
+        /// <param name="peerIpAddress">The IP Address of the sample.</param>
+        private TestSample(bool isInbound, bool expectedIsUsed, bool expectedIsWarningOn, bool expectedIsSyncOff, int expectedTimeOffsetMs, TimeSpan inputTimeOffset, IPAddress peerIpAddress)
+        {
+            this.IsInbound = isInbound;
+            this.ExpectedIsUsed = expectedIsUsed;
+            this.ExpectedIsWarningOn = expectedIsWarningOn;
+            this.ExpectedIsSyncOff = expectedIsSyncOff;
+            this.ExpectedTimeOffsetMs = expectedTimeOffsetMs;
+            this.InputTimeOffset = inputTimeOffset;
+            this.PeerIpAddress = peerIpAddress;
+        }
+        
+        public static TestSample Inbound(bool expectedIsUsedResult, bool expectedIsWarningOn, bool expectedIsSyncOff, int expectedTimeOffsetLessThanMs, TimeSpan inputTimeOffset, IPAddress ipAddress)
```

Comment:
Why in these static methods you use expectedTimeOffsetLessThanMs Instead of expectedTimeOffset